Correct grammar in README.md

danieldegroot2 opened the pull request:
Correct some typos, grammar and spelling in README.md
Also removes redundant and trailing spaces.

danieldegroot2 committed:
Correct grammar in README.md

KhartesViewer commented:
Thanks for taking the time to find all of these problems!  Some general comments, to explain why I may take a while to evaluate your pull request.

1) Most of the grammatical corrections are good, thank you!
2) Thank you for finding the unnecessary spaces at the ends of lines.
3) I haven't decided yet about the changes in headings ('###' to '####', etc); I need to study these some more.
4) Thanks for updating the Discord link.
5) I do not agree with the many cases where two spaces after a period are reduced to one.  I realize that one space is the more modern style, but I am used to typing two spaces, and don't really feel the need to change.  So I will not accept these particular changes.

As I mentioned, it may take me a little while to digest all these changes (other than the two-to-one spaces change, where I have already decided), so I wanted to let you know.  Thanks again for working on this!

danieldegroot2 commented:
> I haven't decided yet about the changes in headings ('###' to '####', etc); I need to study these some more.

This is to avoid using H1 headers, they are usually reserved for the title of the document itself / the page name of a site.
Also, they are quite big, and with smaller headers there is less clutter, so it should be easier to read.
(Btw, there is currently no H2 header for the introduction, but in a GitHub document this is somewhat redundant anyway.)

You can use the Outline button [here](https://github.com/danieldegroot2/khartes/blob/readme-qa/README.md) (bulleted list icon in the top right on the toolbar) to look at the structure.
![image](https://github.com/KhartesViewer/khartes/assets/67521919/03c30f80-7f98-4345-b83f-3b824de73df3)

danieldegroot2 committed:
Reinstate whitespace after full stop, remove misc trailing spaces

danieldegroot2 committed:
Remove misc redundant space, change orderless list to bullet list.

danieldegroot2 committed:
Correct grammar, misc.

Correct grammar, spacing. Write out numbers and capitalize consistently. Remove temporary details. Add links to GitHub files.

danieldegroot2 commented:
@KhartesViewer I have added back the double spaces.

Also, I have changed the spacing elsewhere to reduce hanging text.
(mostly 1-3 words followed by a new line at the start of a sentence)
Let me know if this works for you.

You should try [reviewing proposed changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request) if you haven't yet.

KhartesViewer commented:
Thanks for the info about outline viewer, and for the link to the how-to page on reviewing proposed changes; these are both very useful!  I've been fighting bugs in other branches, but I should be getting back to your PR soon.

danieldegroot2 commented:
No problem. I'm aware you're quite busy :-)

KhartesViewer commented on README.md:
```diff
@@ -203,22 +199,22 @@ the green slice crosses the red slice.  Likewise, in the window of the green dat
-### Navigation by keyboard
+#### Navigation by keyboard
 
-You can use the four arrow key to move the current slice (the one where the mouse cursor is
```
Nice catch!

KhartesViewer commented on README.md:
```diff
@@ -203,22 +199,22 @@ the green slice crosses the red slice.  Likewise, in the window of the green dat
+#### Navigation by keyboard
 
-You can use the four arrow key to move the current slice (the one where the mouse cursor is
+You can use the four arrow keys to move the current slice (the one where the mouse cursor is
```
Nice catch!

KhartesViewer replied:
Nice catch!

KhartesViewer commented on README.md:
```diff
@@ -203,22 +199,22 @@ the green slice crosses the red slice.  Likewise, in the window of the green dat
+You can use the four arrow keys to move the current slice (the one where the mouse cursor is
 currently located).  As when navigating with the mouse, the other slices will change to ensure
-that their mutual interesection point remains in the centers of all 3 pairs of crosshairs.
+that their mutual intersection point remains in the centers of all three pairs of crosshairs.
```
Another nice catch!

KhartesViewer commented on README.md:
```diff
@@ -227,12 +223,12 @@ how this changes as you press page-up and page-down.
 watch as the current slice slowly moves in the indicated direction.
 
-The `a`, `s`, `w`, and `m` keys behave the same as the arrow keys, and `e` and `c` behave the
+The `a`, `s`, `w`, and `d` keys behave the same as the arrow keys, and `e` and `c` behave the
```
And another!  This one is especially significant; before your fix, it gave incorrect instructions to the user.

KhartesViewer commented on README.md:
```diff
@@ -10,44 +10,42 @@ pynrrd to read and write NRRD files (a data format for volume files), and OpenCV
-cloning the repository, making sure you have the proper dependencies 
-(see "anaconda_installation.txt" for a list), and then typing `python khartes.py`.  
+cloning the repository, making sure you have the proper dependencies
+(see [anaconda_installation.txt](anaconda_installation.txt) for a list), and then typing `python khartes.py`.
```
Thanks for adding the link!

KhartesViewer commented on README.md:
```diff
@@ -356,7 +351,7 @@ appear in the list of fragments.
 the fragment.  To add a node, position the cursor in one of the
 Data Slices, hold down the shift key (so the cursor turns into
-crosshairs), and click with the left mouse button.
+crosshairs), and press the left mouse button.
```
Nice improvement!

KhartesViewer commented on README.md:
```diff
@@ -372,22 +367,22 @@ was more awkward than it is now.
-this surface is displayed in map view in the Fragment View.  Of course, if
-the new nodes all form a line, only a line, rather than a surface,
-is shown in the Fragment view.  Once the user creates nodes
+* Each node that the user creates is added to a surface (the fragment);
+this surface is displayed in the map view under the Fragment View.  Of course, if
+the new nodes form a line, rather than a surface,
+it is shown in the Fragment View.  Once the user creates nodes
```
```suggestion
* Each node that the user creates is added to a surface (the fragment);
a map view of this surface is displayed in the Fragment View window.  Of course, if
all the nodes lie in a straight line,
then a line, rather than a surface, is shown in the Fragment View.  Once the user creates nodes
```

danieldegroot2 replied:
```suggestion
* Each node that the user creates is added to a surface (the fragment);
a map view of this surface is displayed in the Fragment View window.
If all the nodes lie in a straight line,
then the Fragment View will display a line rather than a surface.  Once the user creates nodes
```
Sure. However, let's keep it short and to the point; too many commas make it hard to comprehend.
I have re-arranged it and dropped "Of course" as it adds unnecessary detail.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -399,77 +394,72 @@ This topic would preferably be introduced later in the
-a picture of crosshairs.  When this button is light, khartes is in
-normal mode, and when it is dark, khartes is in add-node mode.
+a crosshairs icon.  When this button is enabled, khartes is in
+normal mode, and when it is disabled, khartes is in add-node mode.
```
I'm not sure what to do here.  My original "light" and "dark" are incorrect, if the user is in dark mode.  But I don't know whether users understand "enabled" and "disabled" in this context.

danieldegroot2 replied:
```diff
- a picture of crosshairs.  When this button is light, khartes is in
- normal mode, and when it is dark, khartes is in add-node mode.
+ a crosshairs icon.  When this button is not selected, khartes is in
+ normal mode, and when it is selected, khartes is in add-node mode.
```
I have modified it to use "selected" instead. (fyi suggesting changes is not handled, will commit it manually)
Also, let me know if I need to switch them around. I assume `on = normal mode`.

(edited to include below modification)

KhartesViewer replied:
Looks good, except selected = add-node mode.

danieldegroot2 replied:
Thanks. I did think that'd be the case.

KhartesViewer commented on README.md:
```diff
@@ -399,77 +394,72 @@ This topic would preferably be introduced later in the
 
 But what do you do if you discover that a node is in the wrong
-place?  You have a couple of options.
+place? You have a couple of options.
```
```suggestion
place?  You have a couple of options.
```

danieldegroot2 replied:
Will do.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -399,77 +394,72 @@ This topic would preferably be introduced later in the
 the node, so that the node turns cyan.
 At this point, simply hit the "Delete" or "Backspace" key (either will work).
-But bear this in mind: **There is currently no "undo" in khartes!**
+But bear in mind **deleting a node will remove it permanently**.
```
```suggestion
But bear in mind that **deleting a node will remove it permanently**;
there is no "undo" in khartes!
```

danieldegroot2 replied:
```suggestion
But bear in mind that **deleting a node will remove it permanently**;
it can not be recovered.
```
I have modified it to avoid the 'double negative' "no" and "un-".
Also, let's remove the exclamation mark, as it is exhaustive; the fact it is permanent is already emphasized using **bold** formatting.

KhartesViewer replied:
I agree with the added "that", but I do want to reiterate wherever possible that khartes really does not have an "undo" mode; this is a handy place to do that.  I agree that the exclamation mark is excessive.

danieldegroot2 replied:
```suggestion
But bear in mind that **deleting a node will remove it permanently**;
it can not be recovered, as there is currently no undo action implemented.
```

I have added a mention of the undo action, but I have kept the mention of recovery as it helps to keep it in line with the rest of the tutorial.

I should clarify that I also removed the mention of the undo mode because it is already mentioned in https://github.com/KhartesViewer/khartes#things-to-fix , so it may only be temporary. Removing it avoids having to maintain such details. It could alternatively be mentioned once, in bold, at the start of the tutorial.

The added "that" came from your suggested change, but it does look fine.

KhartesViewer replied:
That looks good

KhartesViewer commented on README.md:
```diff
@@ -485,15 +475,14 @@ top left Data Slice, and see how the node moves in the data slice while you pres
-what will happen when you push the left mouse button.
+As you may have noticed, the cursor often changes its appearance,
+depending on where it is in the slice.  The rule is that the shape
+of the cursor tells you what will happen when you press the left mouse button.
```
Nice improvement!

KhartesViewer commented on README.md:
```diff
@@ -518,18 +507,18 @@ are currently holding down the shift key.  In either case, when you press the le
 
 * **Diagonal two-way arrow**: You will only see this while using the TIFF Loader to create a new
-data volume.  Thi shape indicates that the cursor is near a corner of the
+data volume.  This shape indicates that the cursor is near a corner of the
```
Good catch!

KhartesViewer commented on README.md:
```diff
@@ -539,17 +528,17 @@ The prerequisites are:
 direction to go.  If the cursor is near
 an existing node (the node has turned cyan), the auto-segmentation will start
-at that that node.  Otherwise, it will start at the location of the cursor.
+at that node.  Otherwise, it will start at the location of the cursor.
```
Good!

KhartesViewer commented on README.md:
```diff
@@ -539,17 +528,17 @@ The prerequisites are:
 
-To automatically create new points to the right of the starting point, press
-the right bracket or brace key (`]` or `}`); either key will work.  To create new points to the
+To automatically create new nodes to the right of the first one, press
+the right bracket or brace key (`]` or `}`); either key will work.  To create new nodes to the
```
```suggestion
To automatically create new nodes at and to the right of the starting location,
or to the right of the selected node, press
the right bracket or brace key (`]` or `}`); either key will work.  To create new nodes to the
```

KhartesViewer replied:
I'm open to other suggestions here; the main thing to keep in mind is that the cursor may be near an existing node, in which case the new nodes are created to the right of the existing node; or the cursor may not be near an existing node, in which case a new node is created at the cursor location, and at locations to the right.

danieldegroot2 replied:
```diff
+ The auto-segmentation algorithm will automatically create new nodes, going as far as [...]

- At this point, you only need to choose where to begin, and in which direction to go. [...]
- Otherwise, it will start at the location of the cursor.
+ You only need to choose where to begin, and in which direction to go. [...]
+ Otherwise, it will create a node at the location of the cursor and continue from there.

- To automatically create new nodes to the right of the first one, press the right bracket or brace key (] or }); either key will work.
- To create new nodes to the left, press the left bracket or brace key ([] or {).
+ To start and head to the right, press the right bracket or brace key (] or }).
+ To start and head to the left, press the left bracket or brace key ([] or {) instead.

- The auto-segmentation algorithm will go as far as [...]
```
I have modified it to explain the process first and then give instructions.
(If this change is okay, will commit manually.)

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -559,19 +548,18 @@ Note that it will not venture beyond the area that is visible in the Data Slice.
-a couple of seconds (longer if the surface is large, due to khartes updating
-the surface to account for the new points).  If nothing happens within 10 to 15 seconds,
-check to make sure that you met the prerequisites above.
+Note that the algorithm is quite fast; it often adds new nodes within
```
```suggestion
Note that the algorithm is quite fast; it often adds the new nodes within
```

danieldegroot2 replied:
Will do.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -559,19 +548,18 @@ Note that it will not venture beyond the area that is visible in the Data Slice.
+Note that the algorithm is quite fast; it often adds new nodes within
+a couple of seconds (longer for a larger surface, due to khartes updating
+the surface to account for the new nodes).  If nothing happens within 10 to 15 seconds,
+check to make sure that you meet the prerequisites above.
```
```suggestion
check to make sure that you have met the prerequisites above.
```

danieldegroot2 replied:
Sure, my mistake. I would like to avoid mixed past and present tense, but upon further thought this is fine.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -601,13 +589,13 @@ once they have been initially created on disk.
 shift key will toggle between "normal mode" and "add-node mode".  See the section below
 entitled "Advanced Topic: Control Area: Settings" for information on how to change this
-behavioor.
+behavior.
```
Another good catch!  But you have discovered so many typos that henceforth I will not mention each one.  So a collective "good catch" for all the typos you have discovered but that I haven't called out!

KhartesViewer commented on README.md:
```diff
@@ -601,13 +589,13 @@ once they have been initially created on disk.
-you can delete the node by pressing either of these two keys.  As mentioned above, khartes does
-not have an "undo" function, so think carefully before doing this.
+you can delete the node by pressing either of these two keys.  As mentioned above, deleting a node
+will delete it permanently, so think carefully before doing this.
```
```suggestion
will delete it permanently (khartes does not have an "undo" function),
so think carefully before doing this.
```

danieldegroot2 replied:
```diff
- you can delete the node by pressing either of these two keys.  As mentioned above, khartes does
- not have an "undo" function, so think carefully before doing this.
+ you can delete the node by pressing either of these two keys.  As mentioned above, deleted nodes
+ can not be recovered, so think carefully before doing this.
```

See https://github.com/KhartesViewer/khartes/pull/17#discussion_r1381921243 .

KhartesViewer replied:
Same comment as before; I want to emphasize the general lack of an undo function.

danieldegroot2 replied:
I would suggest to avoid duplicate content. This refers back to the first mention, in which it can already be handled.
See https://github.com/KhartesViewer/khartes/pull/17#discussion_r1382369773

KhartesViewer replied:
That seems fine.

KhartesViewer commented on README.md:
```diff
@@ -673,58 +659,58 @@ If you want to work on such a sheet, turn your volume
 
-*Example of a good start.*  A couple of inline slices (the horizontal lines
-in the Fragment View) and a crossline slice (the vertical line) have been 
+*Example of a good start.* A couple of inline slices (the horizontal lines
```
```suggestion
*Example of a good start.*   A couple of inline slices (the horizontal lines
```

danieldegroot2 replied:
```suggestion
*Example of a good start.*  A couple of inline slices (the horizontal lines
```
It only needs one more space, not two. There was already one there.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -673,58 +659,58 @@ If you want to work on such a sheet, turn your volume
 ![sheet_skip](images/sheet_skip.JPG)
 
-***This is bad!***  The horizontal fibers are not continuous.  This needs to be repaired by
+***This is bad!*** The horizontal fibers are not continuous.  This needs to be repaired by
```
```suggestion
***This is bad!***  The horizontal fibers are not continuous.  This needs to be repaired by
```

danieldegroot2 replied:
Will do.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -733,21 +719,20 @@ more repair work you will have to do later.
 
-**Save often.**  You simply need to type Ctrl-S to save your work; try to remember to do this whenever
-you have added or moved a dozen or so nodes.  The "save" operation is very quick, since only 
+**Save often.** You simply need to type Ctrl-S to save your work; try to remember to do this whenever
```
```suggestion
**Save often.**  You simply need to type Ctrl-S to save your work; try to remember to do this whenever
```

danieldegroot2 replied:
Will do.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -758,18 +743,18 @@ and redrawn.  The bigger the triangles, the longer the recomputation
+So, when segmenting, start in the center of a fragment
 and work your way out, keeping a fairly regular mesh, instead
-of trying to create a huge surface first thing.  This practice
+of trying to create a huge surface.  This practice
```
```suggestion
of immediately trying to create a huge surface.  This practice
```

danieldegroot2 replied:
Will do.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -779,7 +764,7 @@ This tab allows you to control important aspects of your fragments.
 now you will have a chance to study it in detail.
 
-The first thing to notice is that one of the rows is hightlighted with a
+The first thing to notice is than one of the rows is highlighted with a
```
```suggestion
The first thing to notice is that one of the rows is highlighted with a
```

danieldegroot2 replied:
Will do. No idea what happened here.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -944,8 +921,8 @@ However, as shown in the picture above,
 These dots are actually the nodes of a fragment whose only purpose
-is to provide annotation.  Here are the steps to add annotation to
-your fragment view:
+is to provide annotation.  Here are the steps to add an annotation to
```
```suggestion
is to provide annotation.  Here are the steps to add annotation to
```

danieldegroot2 replied:
Will do. I see it is more consistent with this section.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -974,41 +951,37 @@ the `Free Node` size.
+### Exporting fragments
 
-Khartes allows you to export your fragments to [`MeshLab`](https://www.meshlab.net/),
+Khartes allows you to export your fragments to [MeshLab](https://www.meshlab.net/),
```
I'm curious to know why you made this change?  I'm not necessarily against it, just curious to know the reason.  Same with the change to the volume-cartographer link below, and to the subsequent mentions of MeshLab.

danieldegroot2 replied:
I assume these are titles/names of products, not their components.
Products are common 'terms', whereas components are usually used in a computing-related context, i.e. 'code'.
Unless the common name is explicitly `formatted like this`, it is better not to do so.

danieldegroot2 replied:
Marking comment as resolved. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -974,41 +951,37 @@ the `Free Node` size.
 2. In the File menu, select `Export file as mesh...`.
 3. After you press the `Save` button, you will be asked to specify
-an "infill spacing".  Just accept the default value.
+an "infill spacing".  The default value should be fine.
```
I like this wording!

KhartesViewer commented on README.md:
````diff
@@ -1022,18 +995,18 @@ that you used.
 vc_render -v [your volpkg directory] --input-mesh [your .obj file] --output-ppm [the name of the ppm file you want to create]
 ```
-You might need to use `--volume` to specify your volume as well, if your volpkg has more than one.
+You might need to use `--volume` to specify your volume as well, if your `.volpkg` has more than one.
````
```suggestion
You might need to use `--volume` to specify your volume as well, if your volpkg has more than one.
```

KhartesViewer replied:
I'm using volpkg as a noun here, rather than as a file extension.  But I will think about this some more; I might end up changing back to your suggestion.

danieldegroot2 replied:
Sure.

danieldegroot2 replied:
Could you let me know what your choice is here, please?
If you want to change it back to a **noun**, you can also suggest to make the change elsewhere.
(depends on your interpretation of each case)

I would suggest using "`.volpkg` file", adding "file" to make it a noun. It should be more comprehensive.
I assume it is still a file (extension).

KhartesViewer replied:
I like your suggestion, which I will modify to "`.volpkg` directory", since Volume Cartographer (and khartes) somewhat confusingly put file extensions on directory names.  Or if you think "folder" is more understandable than "directory", that is fine with me too.

danieldegroot2 replied:
```diff
- You might need to use `--volume` to specify your volume as well, if your `.volpkg` has more than one.
+ You might need to use `--volume` to specify your volume as well, if your `.volpkg` directory has more than one.
```

Directory is fine.

KhartesViewer commented on README.md:
```diff
@@ -1230,44 +1194,42 @@ remember the changes you made in the previous area
 (and your changes)
 will be saved along with the rest of the project;
-you will not need to re-import the fragment the
+you will not need to re-import the fragment
```
```suggestion
you will not need to re-import the fragment the
```

KhartesViewer replied:
A rare case where I disagree with your suggestion.  I can't give a good grammatical reason, however; it's just a feeling.

danieldegroot2 replied:
Okay.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -1352,13 +1312,13 @@ One thing to keep in mind, especially if you are working over a network, is
-updated.  Though these changes happen quickly, they do require a certain amount of time, and when you have a
-large group of nodes you need to edit, you might find that the updating is delaying your work.
+updated.  Though these changes happen quickly, updates do require a certain amount of time, and when
+you have a large group of nodes you need to edit, you might find that updates will delay your work.
```
```suggestion
you have a large group of nodes you need to edit, you might find that waiting for the updates
is delaying your work.
```

danieldegroot2 replied:
Sure, looks fine.

danieldegroot2 replied:
Unresolving comment. Modifications were confirmed to be okay (see https://github.com/KhartesViewer/khartes/pull/17#issuecomment-1793367671)

KhartesViewer commented on README.md:
```diff
@@ -1377,87 +1337,81 @@ The answer is, "as little as possible".
 download the data over the internet);
 * Permission to use small excerpts from the data set in this README file and the associated videos;
-* A [lively community](https://discord.com/invite/6FgWYNjb4N) of fellow enthusiasts;
+* A [lively community](https://discord.gg/V4fJhvtaQn) of fellow enthusiasts;
```
Thanks for updating the link!

KhartesViewer reviewed (commented):
While reviewing all your changes, I really came to appreciate how much time you must have spent on this.  Thank you for all the work!

danieldegroot2 commented on README.md:
```diff
@@ -733,21 +719,20 @@ more repair work you will have to do later.
 
 *The surface has been repaired;* horizontal fibers are now continuous.  The inline and crossline slices show
 the location of the original (magenta) and repaired (cyan) surfaces.  Note that these overlap in one half
```
I assume these colours are the same regardless of the user being in light or dark mode.
Let me know if this is correct.

KhartesViewer replied:
Yes.  The colors (magenta and cyan) were set by the user, and so do not depend on light/dark mode.  But I'm glad you checked.

danieldegroot2 commented:
(btw, you can ignore branch `readme-qa-2` in my repo, it is only a backup of the changes I made so far.)

KhartesViewer commented:
Almost there!  I marked almost everything "resolved"; so only a couple of unresolved items left.
I suggest that you go ahead and commit everything after you make the changes.  Once you have done this, I'll do a final check.  IF everything looks good, I will merge the PR at that time.
Thanks for all the work!

danieldegroot2 commented:
At least for ease of maintenance, I have to keep conversations with modifications open until I have succesfully commited them.
Sorry for any inconvenience on your side.

(You can 'close' ones you don't need to comment on anymore by clicking on the dark header above them. May require scrolling up and down to check you are not missing any conversations.)

KhartesViewer commented:
Ah, got it!  I am completely new to the github-based code-review workflow, as you may have noticed.  So: it sounds like I should not mark things "resolved" until you have had a chance to resolve them (makes sense, now that I've explained it to myself...):

danieldegroot2 committed:
Apply suggestions from code review

Co-authored-by: KhartesViewer <133787404+KhartesViewer@users.noreply.github.com>

danieldegroot2 committed:
Manually apply suggestions from code review

KhartesViewer reviewed (commented):
I agree with all of your changes (and answered your question about volpkg).  I added comments wherever I could, as a way of verifying that I saw your latest suggestions; let me know if I missed anything.
As you know, I am new at this, so once you have made all the changes, please let me know what the next step is (I understand that the eventual goal is for me to merge your readme-qa branch into my main branch, but I don't know if there is anything else that needs to be done first).
Thanks!

danieldegroot2 committed:
Apply suggestions from code review

danieldegroot2 committed:
Manually apply suggestions from code review

danieldegroot2 commented:
> As you know, I am new at this, so once you have made all the changes, please let me know what the next step is (I understand that the eventual goal is for me to merge your readme-qa branch into my main branch, but I don't know if there is anything else that needs to be done first).

@KhartesViewer You can go ahead and merge. No additional steps needed.

KhartesViewer commented:
Ok, the merging succeeded.  Thanks again for all your work on this!

danieldegroot2 commented:
No problem.
